nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -702,6 +702,43 @@ in {
         '';
       };
     };
+
+    ensureUsers = mkOption {
+      default = {};
+      description = lib.mdDoc ''
+        List of user accounts which get automatically created if they don't
+        exist yet. This option does not delete accounts which are not listed
+        anymore.
```

Comment:
When designing new options, why not make them fully declarative? Dump some state about which users are managed with by NixOS so that they can be removed again when they are removed from the config?